driver/docker-container/driver.go

```diff
@@ -305,20 +305,21 @@ func (d *Driver) Rm(ctx context.Context, force bool, rmVolume bool) error {
 		if err != nil {
 			return err
 		}
-		if err := d.DockerAPI.ContainerRemove(ctx, d.Name, dockertypes.ContainerRemoveOptions{
-			RemoveVolumes: true,
-			Force:         force,
-		}); err != nil {
-			return err
-		}
-		for _, v := range container.Mounts {
-			if v.Name == d.Name+volumeStateSuffix {
-				if rmVolume {
-					return d.DockerAPI.VolumeRemove(ctx, d.Name+volumeStateSuffix, false)
+		if rmBuildkit {
+			if err := d.DockerAPI.ContainerRemove(ctx, d.Name, dockertypes.ContainerRemoveOptions{
+				RemoveVolumes: true,
+				Force:         force,
+			}); err != nil {
+				return err
+			}
+			for _, v := range container.Mounts {
+				if v.Name == d.Name+volumeStateSuffix {
+					if rmVolume {
+						return d.DockerAPI.VolumeRemove(ctx, d.Name+volumeStateSuffix, false)
```

Comment:
Are we sure we return here ?

Reply:
FYI this appears as part of my change but it's just a matter of indentation. If that's a real concern, we may need to open a new issue upsteam and create a new PR. 

Though I've taken a look at this function call. The context is there https://github.com/DataDog/buildx/blob/bcfd4348290ecb67122a882e0ef42aa537a51ae6/vendor/github.com/moby/buildkit/util/appcontext/appcontext.go#L17 and does not seems to have timeout. But there may have some hidden timeouts somewhere in the call stack.